kimchi/src/prover.rs

```diff
@@ -871,6 +935,11 @@ where
                     }),
                     generic_selector: DensePolynomial::eval_polynomial(&es.generic_selector, e1),
                     poseidon_selector: DensePolynomial::eval_polynomial(&es.poseidon_selector, e1),
+                    foreign_mul_selector: es
+                        .foreign_mul_selector
+                        .iter()
+                        .map(|poly| DensePolynomial::eval_polynomial(poly, e1))
+                        .collect(),
```

Comment:
I'm actually not sure I understand why you evaluate these selectors in the proof. Where does the linearization fail if you don't?

Reply:
The comment says it's evaluating the polynomials without chunking them.  They are then used to compute the `ft` polynomial for Maller's optimization.

Reply:
not sure I understand, didn't you add the evaluations of your selectors in the proof?

Reply:
I'm not sure I understand either.  I'm following the pattern that other gates are using, such as generic and poseidon on the proceeding two lines.

Reply:
The generic and poseidon gates are special cases, your gate should mirror the behaviour of the other gates (e.g. varbasemul)

Reply:
OK, now it makes sense!  Thanks 👍🏼

Reply:
to add colors: there's no reason to evaluate something unless you have to (because it gets multiplied with another commitment)